src/Shared/E2ETesting/WaitAssert.cs

```diff
@@ -23,6 +23,59 @@ public static void Equal<T>(this IWebDriver driver, T expected, Func<T> actual)
         public static void True(this IWebDriver driver, Func<bool> actual)
             => WaitAssertCore(driver, () => Assert.True(actual()));
 
+        public static void ExecuteScript(this IWebDriver driver, string script)
+        {
+            if (!(driver is IJavaScriptExecutor javaScript))
+            {
+                Assert.False(true, "The driver can't execute JavaScript.");
+                return;
+            }
+            else
+            {
+                javaScript.ExecuteScript(script);
+            }
+        }
+
+        public static void ExecuteAsyncScript(this IWebDriver driver, string script)
+        {
+            if (!(driver is IJavaScriptExecutor javaScript))
+            {
+                Assert.False(true, "The driver can't execute JavaScript.");
+                return;
+            }
+            else
+            {
+                var scriptWithCallback = $"const cb = arguments[arguments.length - 1];{script}.then(cb, cb)";
+                javaScript.ExecuteAsyncScript(scriptWithCallback);
+            }
+        }
+        public static void HasJavaScriptValue<T>(
+            this IWebDriver driver,
+            T expectedValue,
+            string script,
+            Func<object, T> converter = null)
+        {
+            if(!script.StartsWith("return "))
+            {
+                script = $"return {script}";
+            }
```

Comment:
Do we need this level of cleverness?

Reply:
Yes, because it saves you from a common pitfall. (I spent like 1h on this the other day, couldn't figure out why I was only seeing null)